ui-modules/app-inspector/app/components/entity-tree/entity-tree.directive.js

```diff
@@ -43,19 +43,25 @@ export function entityTreeDirective() {
     return {
         restrict: 'E',
         template: entityTreeTemplate,
-        controller: ['$scope', '$state', 'applicationApi', 'iconService', 'brWebNotifications', controller],
+        controller: ['$scope', '$state', 'applicationApi', 'entityApi', 'iconService', 'brWebNotifications', controller],
         controllerAs: 'vm'
     };
 
-    function controller($scope, $state, applicationApi, iconService, brWebNotifications) {
+    function controller($scope, $state, applicationApi, entityApi, iconService, brWebNotifications) {
         $scope.$emit(HIDE_INTERSTITIAL_SPINNER_EVENT);
 
         let vm = this;
 
         let observers = [];
 
+        let timeData = [];
+
         applicationApi.applicationsTree().then((response)=> {
             vm.applications = response.data;
+            vm.applications.forEach(app => {
+                getTimeData(app);
```

Comment:
I'm wondering if getting the time of the first activity is really necessary. `applicationApi.applicationsTree()` API returns the list of applications already ordered by age (older first). So theoretically, the only thing you need to is add an index to each application object to be able to sort ascending or descending.

Reply:
The list returned from applicationsTree isn't in order - In process of this story I launched three empty software processes labelled 1,2,3 and the order returned by the api was not the same as the launch age.

Reply:
@frogfather Hum, the underlying REST API `/v1/applications/fetch` returns consistently the same order on my local environment. So maybe the javascript code does something else somewhere.

Reply:
@tbouron Odd - before modifying any code the order returned was definitely not the launch order. 
The story was raised because it was noticed that the display order didn't correspond to the launch order.